File: napakalaki/src/Napakalaki/Monster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

/**
 *
 * @author marcos
 */
public class Monster {
    private String name;
    private int combatLevel;
    private Prize price;
    private BadConsequence bc;
    private int levelChangeAgainstCultisPlayer;
    
    
    public Monster (String _name, int level, BadConsequence _bc, Prize _price ){
        name=_name;
        combatLevel=level;
        price=_price;
        bc=_bc;
        levelChangeAgainstCultisPlayer=0;
    }
    
    public Monster(String n, int l, BadConsequence _bc, Prize p, int IC){
        name=n;
        combatLevel=l;
        bc=_bc;
        price=p;
        levelChangeAgainstCultisPlayer=IC;
    }
    
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return bc;
    }
    
    public Prize getPrize(){
        return price;
    }
    
    public int getLevelsGained(){
        return price.getLevel();
    }
    
    public int getTreasuresGained(){
        return price.getTreasure();
    }
    
    public int getCombatLevelAgainstCultistPlayer(){
        return combatLevel+levelChangeAgainstCultisPlayer;
    }
    
    
    @Override
    public String toString(){
        return "Names = " + (name) + "\n "
                + " combatLevels = " + Integer.toString(combatLevel)+"\n "
                + "Prices = " + price.toString()+"\n " 
                + "BadConsequences = " + bc.toString() +"\n " 
                + "\n";
    }
    
    
}

File: napakalaki(ruby)/lib/monster.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module NapakalakiGame
require_relative 'player'

class Monster
  attr_reader :name, :combatLevel, :bc, :price, :levelChangeAgainstCultistPlayer
  def initialize(name, combatLevel, bc, price)
    @name=name
    @combatLevel=combatLevel
    @price=price
    @bc=bc
    @levelChangeAgainstCultistPlayer=0
  end
  
  def Monster.newmonster(n,l,badConsequence, p, ic)
    @levelChangeAgainstCultistPlayer=ic
    Monster.new(n,l,badConsequence, p)
    
  end
  
  def getLevelsGained
    return @price.levels
  end
  
  def getTreasuresGained
    return @price.treasures
  end
  
  def getCombatLevelAgainstCultistPlayer
    return @levelChangeAgainstCultistPlayer+@combatLevel
  end
  
  #def getCombatLevel()
    #return @combatLevel
  #end
  
  def to_s
   "\nMostruo : #{@name} \n Nivel: #{@combatLevel} \n \n Buen rollo: #{@price}  \n Mal rollo: #{@bc} \n Level de cambio Cultist Player: #{@levelChangeAgainstCultistPlayer}"
  end
  
  end
end